src/context.c

```diff
@@ -583,13 +583,30 @@ ly_ctx_unset_options(struct ly_ctx *ctx, uint16_t option)
 
         index = 0;
         while ((mod = (struct lys_module *)ly_ctx_get_module_iter(ctx, &index))) {
+            LY_ARRAY_COUNT_TYPE u, v;
+            struct lysc_ext_instance *exts;
+            struct lysc_ext_substmt *substmts;
+            struct lysc_node *root;
+
             /* ignore modules that didn't compile */
             if (!mod->compiled) {
                 continue;
             }
 
             /* set NULL for all ::lysc_node.priv pointers in module */
             lysc_module_dfs_full(mod, lysc_node_clear_priv_dfs_cb, NULL);
+
+            /* set NULL for all ::lysc_node.priv pointers in extensions */
+            exts = mod->compiled->exts;
+            LY_ARRAY_FOR(exts, u) {
+                substmts = exts[u].substmts;
+                LY_ARRAY_FOR(substmts, v) {
+                    if (LY_STMT_IS_DATA_NODE(substmts[v].stmt)) {
+                        root = *(struct lysc_node **)substmts[v].storage;
```

Comment:
I do not think this is safe, in general. You can do it only if `lys_compile_extension_instance()` was called so it should probably set some flag that will be read here. Based on your other changes and my suggested ones, you can use the presence of `parsed` instead of a flag, since it can be set only by `lys_compile_extension_instance()`, which is the only proper way to get the compiled node structures.